tests/ga/test_monitor.py

```diff
@@ -229,6 +277,158 @@ def test_failed_heartbeat_creates_telemetry(self, patch_report_heartbeat, _, *ar
         monitor_handler.stop()
 
 
+@patch('azurelinuxagent.common.event.EventLogger.add_event')
+@patch('azurelinuxagent.common.osutil.get_osutil')
+@patch("azurelinuxagent.common.protocol.healthservice.HealthService._report")
+@patch("azurelinuxagent.common.protocol.wire.CryptUtil")
+@patch("azurelinuxagent.common.utils.restutil.http_get")
+class TestEventMonitoring(AgentTestCase):
+    def setUp(self):
+        AgentTestCase.setUp(self)
+        self.lib_dir = tempfile.mkdtemp()
+
+        self.event_logger = EventLogger()
+        self.event_logger.event_dir = os.path.join(self.lib_dir, "events")
+
+    def tearDown(self):
+        fileutil.rm_dirs(self.lib_dir)
+
+    def _create_mock(self, test_data, mock_http_get, MockCryptUtil, *args):
+        """Test enable/disable/uninstall of an extension"""
+        monitor_handler = get_monitor_handler()
+
+        # Mock protocol to return test data
+        mock_http_get.side_effect = test_data.mock_http_get
+        MockCryptUtil.side_effect = test_data.mock_crypt_util
+
+        protocol = WireProtocol("foo.bar")
+        protocol.detect()
+        protocol.report_ext_status = MagicMock()
+        protocol.report_vm_status = MagicMock()
+
+        monitor_handler.protocol_util.get_protocol = Mock(return_value=protocol)
+        return monitor_handler, protocol
+
+    @patch("azurelinuxagent.common.protocol.wire.WireClient.send_event")
+    @patch("azurelinuxagent.common.conf.get_lib_dir")
+    def test_collect_and_send_events_with_small_events(self, mock_lib_dir, patch_send_event, *args):
+        mock_lib_dir.return_value = self.lib_dir
+
+        test_data = WireProtocolData(DATA_FILE)
+        monitor_handler, protocol = self._create_mock(test_data, *args)
+        monitor_handler.init_protocols()
+
+        sizes = [1, 2, 3, 4, 5]  # get the powers of 2, and multiple by 1024.
+
+        for power in sizes:
+            size = 2 ** power * 1024
+            self.event_logger.save_event(create_event_message(size))
+        monitor_handler.collect_and_send_events()
+
+        # The send_event call would be called twice - Once when the buffer fills up from 1 to 4, and then for 5.
+
+        self.assertEqual(2, patch_send_event.call_count)
```

Comment:
can we add asserts verifying the arguments  to these 2 calls?

> Once when the buffer fills up from 1 to 4, and then for 5.

Reply:
Changed the test to be more deterministic.